src/net/gateway-web3.ts

```diff
@@ -252,10 +253,24 @@ export class Web3Gateway {
                 return BigNumber.from(result).toNumber()
             })
             .catch(err => {
-                return 0
+                return -1
             })
     }
 
+    /** Request the block number of the Gateway which is currently connected to */
+    public async getBlockNumber(): Promise<void> {
+        if (this.lastBlockNumber) {
+            return Promise.resolve()
+        } else if (!this._provider) {
+            return Promise.resolve()
```

Comment:
@marcvelmer  why this is `promise.resolve()`?How to you actually know if the function failed? It should never throw?

Reply:
That's a good question where I also had several doubts.

When I call `Promise.all()` what I really want to do is calling `Promise.allSettled()`, because I want `lastBlockNumber` value for all GWs in that Promise, no matter if one or several requests fail. The problem is that calling `Promise.reject()` stops the `Promise.all()`, that's why it should never throw.

Because I'm using `Promise.all()` instead of `Promise.allSettled()`, I have found a 'tricky' way of achieving this:

`await Promise.all(web3getBlockNumberCandidates.map(p => p.catch(e => e))).then()`

But I'm not sure this correct and/or an elegant way of solving this... But now we can reject the Promise and still proceed with all lastBlockNumber calls.

Reply:
Ok, thanks to @brickpop I'm using TS 4.3.2 and I can use `Promise.allSettled()` which is the correct way of implementing what we wanted and having the corresponding `Promise.resolve()` or `Promise.reject()` in every situation.

Reply:
This issue it's outdated because the code has changed. Anyway, there is an issue still pending with moving the TS version to 4.3.2